src/sentry/replays/usecases/query/__init__.py

```diff
@@ -163,6 +165,17 @@ def query_using_optimized_search(
             SearchFilter(SearchKey("environment"), "IN", SearchValue(environments)),
         ]
 
+    for i in range(len(search_filters)):
+        sf = search_filters[i]
+        if sf.key.name in VIEWED_BY_ME_KEY_ALIASES:
+            # "viewed_by_me" is not a valid query field.
+            # It's a convenience alias for users without the admin access to lookup ids.
+            if sf.operator not in ["=", "!="]:
+                raise ParseError("Invalid operator specified for viewed_by_me")
+            search_filters[i] = SearchFilter(
+                SearchKey("viewed_by_id"), sf.operator, SearchValue(request_user_id)
```

Comment:
Even though it sholuldn't be possible we need to handle the case where the user_id is null.  Unless you want to enforce non-nullability of the user in the call site (which would also be a fine outcome).